modules/core/src/main/java/org/apache/ignite/internal/commandline/ClusterStateChangeCommand.java

```diff
@@ -0,0 +1,124 @@
+import static org.apache.ignite.internal.commandline.CommandList.SET_STATE;
+import static org.apache.ignite.internal.commandline.CommandLogger.optional;
+import static org.apache.ignite.internal.commandline.CommandLogger.or;
+import static org.apache.ignite.internal.commandline.CommonArgParser.CMD_AUTO_CONFIRMATION;
+
+/**
+ * Command to change cluster state.
+ */
+public class ClusterStateChangeCommand implements Command<ClusterState> {
+    /** New cluster state */
+    private ClusterState state;
+
+    /** Cluster name. */
+    private String clusterName;
+
+    /** {@inheritDoc} */
+    @Override public void printUsage(Logger log) {
+        Map<String, String> params = new LinkedHashMap<>();
+
+        params.put(ACTIVE.toString(), "Activate cluster. Cache updates are allowed.");
+        params.put(INACTIVE.toString(), "Deactivate cluster.");
+        params.put(ACTIVE_READ_ONLY.toString(), "Activate cluster. Cache updates are denied.");
+
+        Command.usage(log, "Change cluster state:", SET_STATE, params, or((Object[])ClusterState.values()), optional(CMD_AUTO_CONFIRMATION));
+    }
+
+    /** {@inheritDoc} */
+    @Override public void prepareConfirmation(GridClientConfiguration clientCfg) throws Exception {
+        try (GridClient client = Command.startClient(clientCfg)) {
+            clusterName = client.state().clusterName();
+        }
+    }
+
+    /** {@inheritDoc} */
+    @Override public String confirmationPrompt() {
+        return "Warning: the command will change state of cluster with name \"" + clusterName + "\" to " + state + ".";
+    }
+
+    /** {@inheritDoc} */
+    @Override public Object execute(GridClientConfiguration clientCfg, Logger log) throws Exception {
+        try (GridClient client = Command.startClient(clientCfg)) {
+            Set<GridClientNode> serverNodes = client.compute().nodes().stream()
+                .filter(n -> !n.isClient() && !n.isDaemon())
+                .collect(toSet());
+
+            Set<GridClientNode> supportedServerNodes = serverNodes.stream()
+                .filter(n -> n.supports(CLUSTER_READ_ONLY_MODE))
+                .collect(toSet());
+
+            if (state == ACTIVE_READ_ONLY && !supportedServerNodes.equals(serverNodes))
+                throw new IgniteException("Not all nodes in cluster supports cluster state " + state);
+
+            if (F.isEmpty(supportedServerNodes))
+                client.state().active(ClusterState.active(state));
+            else
+                client.state().state(state);
+
+            log.info("Cluster state changed to " + state);
+
+            return null;
+        }
+        catch (Throwable e) {
+            log.info("Failed to change cluster state to " +  state);
+
+            throw e;
+        }
+    }
+
+    /** {@inheritDoc} */
+    @Override public void parseArguments(CommandArgIterator argIter) {
+        String s = argIter.nextArg("New cluster state not found.");
```

Comment:
you need to add it under **try {** section

Reply:
Nope. Look at the `CommandArgIterator#nextArg` implementation.

Reply:
who will throw : IllegalArgumentException here ?

Reply:
> who will throw : IllegalArgumentException here ?
CommandArgIterator